diff --git a/base/thread_annotations.h b/base/thread_annotations.h
index bba3ed7c2..330ec7331 100644
--- a/base/thread_annotations.h
+++ b/base/thread_annotations.h
@@ -219,8 +219,11 @@
 // Disables warnings for a single read operation.  This can be used to avoid
 // warnings when it is known that the read is not actually involved in a race,
 // but the compiler cannot confirm that.
-#define TS_UNCHECKED_READ(x) thread_safety_analysis::ts_unchecked_read(x)
+#ifndef TS_UNCHECKED_READ
+#define TS_UNCHECKED_READ(x) libchrome::thread_safety_analysis::ts_unchecked_read(x)
+#endif
 
+namespace libchrome {
 namespace thread_safety_analysis {
 
 // Takes a reference to a guarded data member, and returns an unguarded
@@ -236,5 +237,6 @@ inline T& ts_unchecked_read(T& v) NO_THREAD_SAFETY_ANALYSIS {
 }
 
 }  // namespace thread_safety_analysis
+} // namespace libchrome
 
 #endif  // BASE_THREAD_ANNOTATIONS_H_
