diff --git a/base/message_loop/message_loop.h b/base/message_loop/message_loop.h
new file mode 100644
index 000000000000..cfdbdd1da2ee
--- /dev/null
+++ b/base/message_loop/message_loop.h
@@ -0,0 +1,32 @@
+#ifndef BASE_MESSAGE_LOOP_MESSAGE_LOOP_H_
+#define BASE_MESSAGE_LOOP_MESSAGE_LOOP_H_
+
+#include "base/message_loop/message_loop_current.h"
+#include "base/task/single_thread_task_executor.h"
+
+namespace base {
+
+
+class MessageLoop : public SingleThreadTaskExecutor {
+ public:
+   using Type = MessagePumpType;
+
+   static constexpr Type TYPE_DEFAULT = Type::DEFAULT;
+   static constexpr Type TYPE_UI = Type::UI;
+   static constexpr Type TYPE_CUSTOM = Type::CUSTOM;
+   static constexpr Type TYPE_IO = Type::IO;
+#if defined(OS_ANDROID)
+   static constexpr Type TYPE_JAVA = Type::JAVA;
+#endif  // defined(OS_ANDROID)
+
+  explicit MessageLoop(Type type = Type::DEFAULT) : SingleThreadTaskExecutor(type) {}
+};
+
+class MessageLoopForIO : public MessageLoop {
+ public:
+  MessageLoopForIO() : MessageLoop(MessagePumpType::IO) {}
+};
+
+}
+
+#endif // BASE_MESSAGE_LOOP_MESSAGE_LOOP_H_
diff --git a/base/threading/thread.h b/base/threading/thread.h
index f19f328916ad..8b0a171ed9bb 100644
--- a/base/threading/thread.h
+++ b/base/threading/thread.h
@@ -13,6 +13,7 @@
 #include "base/base_export.h"
 #include "base/callback.h"
 #include "base/macros.h"
+#include "base/message_loop/message_loop.h"
 #include "base/message_loop/message_pump_type.h"
 #include "base/message_loop/timer_slack.h"
 #include "base/sequence_checker.h"
