diff --git a/base/time/time.cc b/base/time/time.cc
index 54e2035f5ff6..72a5427e423a 100644
--- a/base/time/time.cc
+++ b/base/time/time.cc
@@ -395,12 +395,12 @@ bool Time::FromMillisecondsSinceUnixEpoch(int64_t unix_milliseconds,
 }
 
 int64_t Time::ToRoundedDownMillisecondsSinceUnixEpoch() const {
   // Adjust from Windows epoch (1601) to Unix epoch (1970).
-  int64_t microseconds = us_ - kTimeTToMicrosecondsOffset;
+  int64_t microseconds = base::ClampSub(us_, kTimeTToMicrosecondsOffset);

   // Round the microseconds towards -infinity.
   if (microseconds >= 0) {
     // In this case, rounding towards -infinity means rounding towards 0.
     return microseconds / kMicrosecondsPerMillisecond;
   } else {
     return (microseconds + 1) / kMicrosecondsPerMillisecond - 1;
   }
 }
 
