diff --git a/mojo/public/tools/bindings/generators/mojom_cpp_generator.py b/mojo/public/tools/bindings/generators/mojom_cpp_generator.py
index a0ce5540841d..cf17587cc2a8 100644
--- a/mojo/public/tools/bindings/generators/mojom_cpp_generator.py
+++ b/mojo/public/tools/bindings/generators/mojom_cpp_generator.py
@@ -202,7 +202,7 @@ class StructConstructor(object):
 
   def __init__(self, fields, params):
     self._fields = fields
-    self._params = set(params)
+    self._params = list(params)
 
   @property
   def params(self):
diff --git a/mojo/public/tools/bindings/generators/mojom_mojolpm_generator.py b/mojo/public/tools/bindings/generators/mojom_mojolpm_generator.py
index 4d7ae2b583b0..42e1b11a203e 100644
--- a/mojo/public/tools/bindings/generators/mojom_mojolpm_generator.py
+++ b/mojo/public/tools/bindings/generators/mojom_mojolpm_generator.py
@@ -4,9 +4,9 @@
 """Generates C++ source files from a mojom.Module."""
 import os
 import sys
-from mojom_cpp_generator import _NameFormatter as CppNameFormatter
-from mojom_cpp_generator import Generator as CppGenerator
-from mojom_cpp_generator import IsNativeOnlyKind, NamespaceToArray
+from generators.mojom_cpp_generator import _NameFormatter as CppNameFormatter
+from generators.mojom_cpp_generator import Generator as CppGenerator
+from generators.mojom_cpp_generator import IsNativeOnlyKind, NamespaceToArray
 import mojom.generate.generator as generator
 import mojom.generate.module as mojom
 import mojom.generate.pack as pack
